src/sliceDataStorage.cpp

```diff
@@ -240,7 +175,7 @@ bool SliceMeshStorage::getExtruderIsUsed(const size_t extruder_nr, const LayerIn
     {
         for (const SliceLayerPart& part : layer.parts)
         {
-            if (part.insets.size() > 0 && part.insets[0].size() > 0)
+            if ((!part.wall_toolpaths.empty() && part.hasWallAtInsetIndex(0)) || (!part.spiral_insets.empty() && !part.spiral_insets[0].empty()))
```

Comment:
`hasWallAtInsetIndex(0)` already includes checking if `wall_toolpaths` is empty.
```suggestion
            if(part.hasWallAtInsetIndex(0) || (!part.spiral_insets.empty() && !part.spiral_insets[0].empty()))
```